### PR TITLE
Add citar-capf

### PR DESCRIPTION
**Pull request by mclearc**
Lemme know if this is what you had in mind. I didn't make any changes to the README -- I figured you would want to do that.

### CODE REVIEW

File: citar.el [hunk 1]
```diff
@@ -914,6 +914,71 @@ FORMAT-STRING."
...
 
+;;; Completion at Point Function
+
+(defvar citar-capf--extra-properties
+  (list
+   :exit-function
+   (lambda (str _status)
+     ;; take completion str and replace with key
+     (delete-char (- (length str)))
+     (insert (cadr (assoc str candidates))))
```
Comment (bdarcus) [inline citar.el:925]: I'm looking at this one my phone, but is this right?

Not `gethash` here?

**Comment by bdarcus** [comment]
Merged; thanks again!

**Comment by minad** [comment]
> The main one from a design POV is that the integrated function relies on different packages for different major modes.
> But citar clearly separate those into separate files, that are only auto-loaded as needed.
> WDYT about breaking this apart, and using the "major mode adapter" system?

Indeed, there are issues. But I think going via an extra adapter system is a bit overengineered, which I would avoid. The function just needs a few tweaks here and there. I'll add comments.

**Comment by bdarcus** [comment]
> I think going via an extra adapter system is a bit overengineered.

It wouldn't be an extra "system"; it would make use of the existing system.

Like, there's already a "key-at-point" major mode function.

I don't have in mind, to be clear, this is a good idea. It's just an obvious question to me, given that the system was aimed at addressing this sort of issue.

**Comment by mclearc** [comment]
I think the function is such that `org-element-at-point` will only be called when in org mode, and that is the only dependency. So the suggested changes seem the simplest. But totally up to you obviously!

**Comment by bdarcus** [comment]
> think the function is such that `org-element-at-point` will only be called when in org mode, and that is the only dependency. So the suggested changes seem the simplest. But totally up to you obviously!

In that, seems we're fine just declaring the function, per @minad's point.

**Comment by mclearc** [comment]
> > think the function is such that `org-element-at-point` will only be called when in org mode, and that is the only dependency. So the suggested changes seem the simplest. But totally up to you obviously!
> 
> In that case, seems we're fine just declaring the function, per @minad's point.

Yeah I had this in the original package -- just forgot to include it in the pull request....whoops. :)

**Comment by bdarcus** [comment]
I force-pushed what should mostly fix the issues, but I think the extra-properties part might be wrong.

At least, it won't work (though gives no errors) for me. But my doom/corfu install is messed up, as you know, so ...

File: citar.el [hunk 2]
```diff
@@ -914,6 +914,71 @@ FORMAT-STRING."
...
+               (or (eq ?@ (char-before))
+                   (looking-back capf-citar-markdown-regexp
+                                 (line-beginning-position))))))
+
+      ;; Get and insert candidate
+      (let* ((candidates (citar--ref-completion-table))
+             (begin (save-excursion (backward-word) (point)))
+             (end (point)))
+        (list begin end candidates
+              citar-capf--extra-properties
```
Comment (minad) [inline citar.el:978]: The properties are not specified correctly. Take a look at cape!

**Comment by bdarcus** [comment]
See #631 for a followup PR to finalize this.

**Pull request merged**